MdeModulePkg/Include/Guid/DxeMemoryProtectionSettings.h

```diff
@@ -83,6 +82,10 @@ typedef struct {
   //  FALSE - UEFI Stack Guard will be disabled.
   BOOLEAN                                   CpuStackGuard;
 
+  // If enabled, all EfiConventionalMemory will be marked with EFI_MEMORY_RO. This can
+  // be used in conjunction with the NX setting for EfiConventionalMemory.
+  BOOLEAN                                   MarkFreeMemoryReadOnly;

```

Comment:
Two things:
- This should be `MarkFreeMemoryReadProtected` not ReadOnly.
- I thought we were not having this configurable via the HOB? I.e. you just got a system with RP on free memory because we at the core level decided that was best for you? It wouldn't surprise me if then in compatibility mode we need to disable it. What are your thoughts here?

Reply:
Updated to FreeMemoryReadProtected